src/merge.rs

```diff
@@ -81,7 +81,7 @@ pub fn merge<T, F: FnMut(&T, &T) -> bool>(
             let left = &mut self.start;
             let out = &mut self.dst;
 
-            while *left != self.end && right as *const T != right_end {
+            while *left != self.end && !std::ptr::eq(right, right_end) {
```

Comment:
Please leave this and other equality checks as they are and focus on fixing the build.